src/Compilers/Core/Portable/InternalUtilities/StringExtensions.cs

```diff
@@ -49,6 +49,20 @@ public static bool LooksLikeTypeParameterName(this string name)
             return name.Length >= 3 && name[0] == 'T' && char.IsUpper(name[1]) && char.IsLower(name[2]);
         }
 
+        /// <summary>
+        /// Given an input string changes it to be acceptable as a part of a type name.
+        /// For now we will simply replace '.' with '_'as the most common case.
+        /// </summary>
+        internal static string MangleForTypeNameIfNeeded(this string original)
```

Comment:
This should also be used at https://github.com/dotnet/roslyn/blob/bf91b12a9c6d8fc0683c8fa63869978e5b2a822d/src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeManager.Templates.cs#L296